geoh5py/shared/conversion/grid2d.py

```diff
@@ -194,17 +204,74 @@ def convert_to_pillow(cls, data: np.ndarray) -> Image:
 
         raise IndexError("Only 1, 3, or 4 layers can be selected")
 
+    @classmethod
+    def compute_vertices(cls, input_entity: Grid2D) -> np.ndarray:
+        """
+        Compute the vertices of the geoimage to create based on the
+        properties of the grid and its angle.
+        :param input_entity: The grid2D object to convert.
+        :return: A numpy array of 4 points in x,y,z.
+        """
+        # get the origin in x and y
+        origin_x = input_entity.origin["x"]
+        origin_y = input_entity.origin["y"]
+        origin_z = input_entity.origin["z"]
+
+        # get the length of the x and y axis
+        if (
+            input_entity.u_count is None
+            or input_entity.v_count is None
+            or input_entity.u_cell_size is None
+            or input_entity.v_cell_size is None
+        ):
+            raise AttributeError(
+                "The grid2D object must have the following properties"
+                ": u_count, v_count, u_cell_size, v_cell_size"
+            )
+
+        length_u = input_entity.u_count * input_entity.u_cell_size
+        length_v = input_entity.v_count * input_entity.v_cell_size
+
+        # get the rotation
+        orientation = input_entity.rotation
+
+        # Define the corners
+        corners = np.array([[0, length_v], [length_u, length_v], [length_u, 0], [0, 0]])
+
+        # Define the rotation matrix
+        angle_rad = -np.radians(orientation)
+        rotation_matrix = np.array(
+            [
+                [np.cos(angle_rad), -np.sin(angle_rad)],
+                [np.sin(angle_rad), np.cos(angle_rad)],
+            ]
+        )
+
+        # Rotate and shift the corners
+        rotated_corners = corners @ rotation_matrix
+        shifted_corners = rotated_corners + [origin_x, origin_y]
```

Comment:
We could have a new utils to rotate coordinates about a point, instead of duplicating code here and in grid2d.